### ISSUE TITLE
save after delete???

### ISSUE THREAD

**Issue by xidui** [Automattic/mongoose#3227]
1.I got a document (A) by id(XXX)
2.I remove document by id(XXX)
3.I make a change to A
4.I save A

the phenomenon is that:
1. the document is removed eventually.(I am satisfied with this result)
2.after I save a, it won't make a mistake, I got the 'err' to be 'null' (I am not agree with this result)

**Comment by vkarpov15** [comment]
You're right, currently mongoose will not treat it as an error if it tries to update a document and no documents were actually updated unless there was a version bump. This is definitely something worth thinking about more.

As a workaround, check if the `numAffected` param to save callback is 0.

`doc.save(function(err, doc, numAffected) { /* numAffected will be 0 if no docs were updated */ }));`
